core/app/models/spree/product.rb

```diff
@@ -168,11 +168,13 @@ def deleted?
     end
 
     # Determines if product is available. A product is available if it has not
-    # been deleted and the available_on date is in the past.
+    # been deleted, the available_on date is in the past
+    # and the available_on date is nil or in the future.
```

Comment:
I think we need to change this second `available_on` to `discontinue_on`.

Reply:
Thanks for the early feedback :)

This PR is still in Draft, though.